src/main/kotlin/uk/gov/dwp/dataworks/kafkatohbase/reconciliation/repositories/impl/MetadataStoreRepositoryImpl.kt

```diff
@@ -99,28 +100,46 @@ class MetadataStoreRepositoryImpl(private val connectionSupplier: ConnectionSupp
                     "unit" to trimReconciledUnit,
                     "table" to table,
                     "deleted_count" to deletedCount.toString())
+
                 deletedCount
             }
 
         }
 
-    override fun deleteAllReconciledRecords(): Int =
-        connection().use { connection ->
+    override fun deleteAllReconciledRecords(deletedAccumulation: Int): Int {
+        val deletedCount = connection().use { connection ->
             connection.createStatement().use {
                 val (deletedCount, duration) = measureTimedValue {
-                    it.executeUpdate("""
+                    it.executeUpdate(
+                        """
                         DELETE FROM $table
-                        WHERE reconciled_result = TRUE
-                    """.trimIndent())
+                        WHERE reconciled_result = TRUE 
+                        LIMIT $deleteLimit
+                    """.trimIndent()
+                    )
                 }
 
-                logger.info("Deleted records in Metadata Store", "table" to table,
-                    "deleted_count" to "$deletedCount", "time_taken" to "$duration")
+                logger.info(
+                    "Deleted records in Metadata Store", "table" to table,
+                    "deleted_count" to "$deletedCount", "time_taken" to "$duration", "delete_limit" to "$deleteLimit",
+                    "deleted_accumulation" to "${deletedAccumulation + deletedCount}"
+                )
+
+                if (!connection.autoCommit) {
+                    connection.commit()
+                }
 
                 deletedCount
             }
         }
 
+        if (deletedCount == 0) {
```

Comment:
This should really be: if deleted count is less than the delete batch size

Reply:
Why?

Reply:
I see your point I will push a change.